mentorship ios/Networking/RelationRequestActions.swift

```diff
@@ -40,18 +49,15 @@ class RelationRequestActionAPI: ObservableObject {
         }
         
         //api call
-        cancellable = NetworkManager.callAPI(urlString: urlString, httpMethod: httpMethod, token: token)
+        cancellable = NetworkManager.callAPI(urlString: urlString, httpMethod: httpMethod, token: token, session: urlSession)
             .receive(on: RunLoop.main)
             .catch { _ in Just(self.response) }
-            .sink { [weak self] in
-                self?.response = $0
-                if NetworkManager.responseCode == 200 {
-                    self?.success = true
-                }
+            .sink {
+                completion($0)
         }
     }
 }
 
-struct ResponseMessage: Decodable {
+struct ResponseMessage: Codable {
     let message: String?
```

Comment:
as far as I understand it can also fail with the message ` .catch { _ in Just(self.response) }` so maybe we should have success flag?

Reply:
I don't completely understand this point.
Catch operator assigns the provided value in case of network error. I guess, while mocking we assume the network request completes successfully, right?

Reply:
But this is not only used for mocking. Ones you moved your NetworkStatus == 200 from your view, you'll understand, why.

Reply:
I'm still not clear about this. The catch operator only comes into play when the network request fails (i.e. wi-fi unavailable). Otherwise, it doesn't matters what the response code is, the backend returns the model and we use that.

Reply:
This is what I have shown in my example PR.

Reply:
I'm not able to find any such modification in your example PR. Can you refer me to the specific file and line of code in that PR?

Reply:
It can be replaced by protocol which not return network status but either success or fail with the message.